src/uct/sm/scopy/base/scopy_ep.c

```diff
@@ -145,34 +159,28 @@ ucs_arbiter_cb_result_t uct_scopy_ep_progress_tx(ucs_arbiter_t *arbiter,
                                                 arbiter);
     uct_scopy_ep_t *ep       = ucs_container_of(ucs_arbiter_elem_group(elem),
                                                 uct_scopy_ep_t, arb_group);
+    uct_scopy_tx_t *tx       = ucs_container_of(elem, uct_scopy_tx_t,
+                                                arb_elem);
     unsigned *count          = (unsigned*)arg;
-    uct_scopy_tx_arb_elem_t *tx_arb_elem;
-    uct_scopy_tx_t *tx;
-    uct_scopy_tx_comp_t *flush_comp;
     ucs_status_t status;
     size_t seg_size;
 
-    tx_arb_elem = ucs_container_of(elem, uct_scopy_tx_arb_elem_t, arb_elem);;
-    if (tx_arb_elem->op != UCT_SCOPY_TX_FLUSH_COMP) {
-        ucs_assert((tx_arb_elem->op == UCT_SCOPY_TX_GET_ZCOPY) ||
-                   (tx_arb_elem->op == UCT_SCOPY_TX_PUT_ZCOPY));
-        tx       = ucs_container_of(elem, uct_scopy_tx_t, super.arb_elem);
+    if (tx->op != UCT_SCOPY_TX_FLUSH_COMP) {
+        ucs_assert((tx->op == UCT_SCOPY_TX_GET_ZCOPY) ||
+                   (tx->op == UCT_SCOPY_TX_PUT_ZCOPY));
         seg_size = iface->config.seg_size;
         status   = iface->tx(&ep->super.super, tx->iov, tx->iov_cnt,
                              &tx->iov_iter, &seg_size, tx->remote_addr,
-                             tx->rkey, tx->super.op);
-        status = uct_scopy_ep_comp_tx(ep, tx, status, seg_size);
+                             tx->rkey, tx->op);
+        status = uct_scopy_ep_tx_comp(ep, tx, status, seg_size);
```

Comment:
why do we need to pass status and return same status from uct_scopy_ep_tx_comp?
can we refactor the code so we do 'if (status-is-err) { return UCS_ARBITER_CB_RESULT_REMOVE_ELEM }' immediately after uct_scopy_ep_tx_comp, then we can remove the function uct_scopy_ep_tx_comp altogether and move its code to here? and keep only "uct_scopy_ep_tx_comp_common" which would be called outside the 'if' after line 184

Reply:
done